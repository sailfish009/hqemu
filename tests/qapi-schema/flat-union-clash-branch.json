# Flat union branch name collision
# FIXME: this parses, but then fails to compile due to a duplicate 'c_d'
# (one from the base member, the other from the branch name).  We should
# either reject the collision at parse time, or munge the generated branch
# name to allow this to compile.
{ 'enum': 'TestEnum',
  'data': [ 'base', 'c-d' ] }
{ 'struct': 'Base',
  'data': { 'enum1': 'TestEnum', '*c_d': 'str' } }
{ 'struct': 'Branch1',
  'data': { 'string': 'str' } }
{ 'struct': 'Branch2',
  'data': { 'value': 'int' } }
{ 'union': 'TestUnion',
  'base': 'Base',
  'discriminator': 'enum1',
  'data': { 'base': 'Branch1',
            'c-d': 'Branch2' } }
