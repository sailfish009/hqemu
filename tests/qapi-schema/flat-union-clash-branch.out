object :empty
object Base
    member enum1: TestEnum optional=False
    member c_d: str optional=True
object Branch1
    member string: str optional=False
object Branch2
    member value: int optional=False
enum TestEnum ['base', 'c-d']
object TestUnion
    base Base
    tag enum1
    case base: Branch1
    case c-d: Branch2
